src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/Models/RecordingFrameStateData.cs

```diff
@@ -14,7 +14,7 @@ namespace RegressionGames.StateRecorder.Models
     public class RecordingFrameStateData
     {
         //Update me if fields/types change
-        public int apiVersion = SdkApiVersion.VERSION_4;
+        public int apiVersion = SdkApiVersion.VERSION_12;
```

Comment:
Question for maybe @RG-nAmKcAz and @abeizer - Are we syncing the API version across all of the data types? If so, why not define it in one place? I suppose I am still a little confused on the versioning strategy here.

Reply:
There is one class that has the current api version.  Any class that is updated gets a reference to the API version on which it was updated, thus when it persists it writes out that api version.

If you look in the bot segment recordings you will see apiVersion in nearly every sub object of the json.  When the json is parsed, so long as your current runtime api version is >= the highest apiVersion found in the file, then you are good, else you will get a warning that you can't parse that newer api version file.

This strategy allows for full backwards compatibility.  It also allows some level of forwards compatibility in that recordings or bot segments created on newer api version SDKs will still load on the older version as long as no changed/new features are used.  Since the version is tracked on a feature specific level throughout this is possible.